README.md
The principal purpose of this addon is help the player to chose the best item for is character.
GearHelper can show you if an item is better than the one equipped depending on custom weights or Noxxic templates.
Gearhelper can also automatically needor greeddepending what is best for the player, or automatically validate questtaking the best item, or the most expensive.
In raid, if a player loot an item that is better than what you have equipped, GearHelper will tel you so. But unlike Personal Loot Helper, GearHelperwill offer you to send him a message, in his language, simply by clicking on his name in the message print by GearHelper.

Videos : https://www.youtube.com/watch?v=NQsBeBzhgBU

    •	Cmd
    •	/ghou /gh config—> Open GearHelper configuration panel
    •	/gh cw—> Open Custom Weights configuration panel
    •	Debug command
    •	/gh version —> Print you the actual version of GearHelper
    •	/gh ram —> Print you the ram used by GearHelper
    •	Functionnalities :
    •	Calculate if a looted stuff is better than equipped one
    •	Print a message "This item is better" or "This item is worse" on tooltip
    •	Print a green dot in bags if an item is better
    •	Tooltip border colored in green or red
    •	Possibility to compute ilvl
    •	Possibility to compute empty sockets
    •	Print you a message in raid if someone loot something better than what you have, allowing you to ask him if he needs, easily
    •	Automatically sell grey items
    •	Automatically greed in instances
    •	Automatically need in instances
    •	Automatically accept quest reward
    •	If this option is turned off, print a green dot on the best item, or gold coins on the most expensive
    •	Automatically equip looted stuff if needed
    •	Automatically equip most suitable stuff after a specialization changed
    •	Custom Weights
    •	Possibility to write them as percentage
    •	Automatically repair
    •	Disabled
    •	Repair with your own golds
    •	Repair with guild founds, or own golds if your guild don’t allow you to repair
    •	Automatically invite a player that whippet you a predefined message (default : +GH123-). Allow you to create group easily for world bosses…
    •	Play a sound when a player is whispering you
    •	Show an alert and play a sound when a player call your name
    •	Other :
    •	Profil by character
    •	Check with other GearHelper if you’re up to date, and tell you if not

    •	Trello GearHelper (French)

---

Le but principal de cet addon est d’aider le joueur à choisir l’item le plus adapté à son personnage.
GearHelper peut afficher si un item est mieux que celui équipé en fonction de template personnalisé, ou d’un template par défaut pour sa spé.
GearHelper peut également faire besoinou cupiditéautomatiquementen fonction de ce qui est mieux pour le joueur, ou encore valider automatiquement les quêtesen prenant le stuff le plus adapté ou celui qui se vendra à un meilleur prix.
En raid, si un joueur de votre groupe loot un item qui serait une amélioration pour vous, GearHelpervous l’indiquera. Mais contrairement à Personnal Loot Helper, GearHelpervous propose d’envoyer un message au joueur dans sa langue, simplement en cliquant sur le nom du joueur, dans le message proposé par GearHelper.

    •	Cmd
    •	/ghou /gh config—> permet d’ouvrir le panneau de configuration de GearHelper
    •	/gh cw—> permet d’ouvrir le panneau de configuration des Custom Weights (templates personnalisés)

    •	Debug command
    •	/gh version —> Affiche votre version actuelle de GearHelper
    •	/gh ram —> Affiche la quantité de rat utilisée par GearHelper

    •	(tooltip = bulle d’aide)
    •	Fonctionnalités :
    •	Calculer si un stuff loot est mieux que celui équipé
    •	Affichage d’un message "cet item est mieux" ou "cet item est moins bien" sur le tooltip
    •	Affichage d’un point vert dans les sacs si l’item est mieux
    •	Contour du tooltip vert ou rouge en fonction du calcul
    •	Possibilité de prendre en compte l’ilvl dans le calcul et d’indiquer le facteur d’ilvl
    •	Possibilité de prendre en compte les chasses de gemme vides dans le calcul
    •	Vous affiche un message en raid si quelqu’un loot quelque chose de mieux que ce que vous avez et vous permet de lui demander s’il en a besoin, facilement
    •	Vente automatique des objets gris
    •	Cupidité automatique en donjon
    •	Besoin automatique en donjon
    •	Validation automatique des quêtes
    •	Si cette option est désactivée, affiche un point vert sur l’item le mieux pour vous, ou une icône de pièces d’or sur celui qui se vendra au meilleur prix
    •	Equiper automatiquement le stuff looté si besoin
    •	Equiper automatiquement le stuff le plus adapté après un changement de spé
    •	Possibilité d’afficher un message quand un stuff est équipé automatiquement
    •	Custom Weights / Templates Personnalisés
    •	Possibilité de les renseigner normalement
    •	Possibilité de les renseigner en pourcentage
    •	Réparation automatique
    •	Désactivé
    •	Réparer avec l’argent personnel
    •	Réparer avec l’argent de la guilde, ou les fonds personnel si la guilde ne le permet pas
    •	Invitation automatique des joueurs qui vous chuchotent un message prédéfinis (par défaut : +GH123-) ce qui permet de former des groupes facilement pour les world boss / attaques des capitales…
    •	Joue un son quand un joueur vous /w
    •	Alerte visuelle et sonore quand votre nom est cité
    •	Autres :
    •	Profile par personnage
    •	Recherche des mises à jours chez les autres GearHelper et vous l’indique
    •	Future maj :
    •	Auto dez plutôt que cupi / need

    •	Trello GearHelper (French)

---


CHANGES.txt
lightweight tag 8dbebf3888878855f50b18949fd33675b0b32cdd 2.3
Author:	GitHub <noreply@github.com>
Date:	Fri Apr 5 16:51:07 2019 +0200

commit e4e143b26587537b755ab49de4522c30f3e26e0e
Author: Raphael Daumas <raphdaumas@hotmail.fr>
Date:   Mon Mar 18 13:13:17 2019 +0100

    Update WeightValues.lua

commit e4e143b26587537b755ab49de4522c30f3e26e0e
Author: Raphael Daumas <raphdaumas@hotmail.fr>
Date:   Mon Mar 18 13:13:17 2019 +0100

    Update WeightValues.lua

commit e4e143b26587537b755ab49de4522c30f3e26e0e
Author: Raphael Daumas <raphdaumas@hotmail.fr>
Date:   Mon Mar 18 13:13:17 2019 +0100

    Update WeightValues.lua

commit e4e143b26587537b755ab49de4522c30f3e26e0e
Author: Raphael Daumas <raphdaumas@hotmail.fr>
Date:   Mon Mar 18 13:13:17 2019 +0100

    Update WeightValues.lua

commit 60310e60a920af5de86337bee2e7a6cb173517b9
Author: Mantisis <170255@supinfo.com>
Date:   Fri Mar 8 21:55:19 2019 +0100

    Fix holdable weapons

commit 3989db2c95c08dbc56a380e18daec1142f090121
Author: Mantisis <170255@supinfo.com>
Date:   Fri Mar 8 21:36:21 2019 +0100

    Refacto QuestReward and AskNeed

commit 91466d7821c5135310cc978c78706cd39ae13170
Author: Mantisis <170255@supinfo.com>
Date:   Thu Mar 7 22:37:44 2019 +0100

    remove greed and need options

commit f60e245cd90214ad5cde7ca6fe5e4e507438e652
Author: Mantisis <170255@supinfo.com>
Date:   Thu Mar 7 22:30:06 2019 +0100

    Remove auto need & greed

commit b3f6a4647d08c13f2d2c8fed78ca3b427c485ce6
Author: Mantisis <170255@supinfo.com>
Date:   Thu Mar 7 21:17:28 2019 +0100

    equipItem refacto

commit dfbad8c27ccd684d2d7bd22ff300150086a19d60
Author: Mantisis <170255@supinfo.com>
Date:   Wed Mar 6 23:29:16 2019 +0100

    Remove print

commit b59dc8922f1cd336d194b91e8b7c11d3b272a834
Author: Mantisis <170255@supinfo.com>
Date:   Wed Mar 6 23:25:44 2019 +0100

    Handle main weapon by class and spec

commit 5762a5ce9d785cff6bfa2a00b66663d73cab1348
Author: Mantisis <170255@supinfo.com>
Date:   Sun Mar 3 01:19:04 2019 +0100

    Fix 2H weapon calculation

commit 5762a5ce9d785cff6bfa2a00b66663d73cab1348
Author: Mantisis <170255@supinfo.com>
Date:   Sun Mar 3 01:19:04 2019 +0100

    Fix 2H weapon calculation

commit 0c17c19352d244ccbb2ace8e343e868d476e5e87
Author: Raphael Daumas <raphdaumas@hotmail.fr>
Date:   Fri Mar 1 09:25:59 2019 +0100

    Update .gitignore

commit 80135beeed649b6d2ea34af2555dfc6c5140f5c8
Author: Mantisis <170255@supinfo.com>
Date:   Wed Feb 27 23:27:19 2019 +0100

    Align content and remove comments

commit 437c1404cb39fa161193d0a954f29f0363c47c52
Author: Mantisis <170255@supinfo.com>
Date:   Wed Feb 27 22:58:33 2019 +0100

    Add back color, tooltip lines etc..

commit 06533b559e8d755e12637919e7b86af1dc77c28b
Author: Mantisis <170255@supinfo.com>
Date:   Wed Feb 27 20:55:26 2019 +0100

    Refacto newweightcalc



GH_Events.lua
local L = LibStub("AceLocale-3.0"):GetLocale("GearHelper")

local gagne = 0
local lfrCheckIsChecked = false
local lastBagUpdateEvent = time()
-- waitingIDTable = waitingIDTable

local function AddonLoaded(_, _, name)
	GearHelper:BenchmarkCountFuncCall("AddonLoaded")
	if GearHelper.db.global.templates == nil then
		GearHelper.db.global.templates = {}
	end

	GearHelper:InitTemplates()

	if name ~= addonName then
		do
			return
		end
	end

	print(GearHelper:ColorizeString(L["merci"], "LightGreen"))
	local runningBuild = select(4, GetBuildInfo())
	if GearHelper.db.global.buildVersion ~= runningBuild then
		GearHelper.db.global.buildVersion = runningBuild
		GearHelper:ResetCache()
	end
end

local function OnMerchantShow()
	GearHelper:BenchmarkCountFuncCall("OnMerchantShow")
	gagne = 0
	if GearHelper.db.profile.sellGreyItems then
		for bag = 0, 4 do
			for slot = 1, GetContainerNumSlots(bag) do
				if GetContainerItemID(bag, slot) ~= nil then
					local id = GetContainerItemID(bag, slot)
					if id then
						local result = GearHelper:SiObjetGris(id)
						if result[1] then
							UseContainerItem(bag, slot)
							gagne = gagne + result[2]
						end
					end
				end
			end
		end
	end

	if CanMerchantRepair() and GearHelper.db.profile.autoRepair == 1 or GearHelper.db.profile.autoRepair == 2 then
		local argentPossedee = GetMoney()
		local prix = GetRepairAllCost()
		local droitGuilde = ""
		local argentGuilde = ""

		if IsInGuild() and CanGuildBankRepair() then
			droitGuilde = GetGuildBankWithdrawMoney()
			argentGuilde = GetGuildBankMoney()
		end
		if prix > 0 then
			if GearHelper.db.profile.autoRepair == 1 then
				if argentPossedee >= prix then
					RepairAllItems(false)
					print(GearHelper:ColorizeString(L["repairCost"], "Pink") .. math.floor(prix / 10000) .. L["dot"] .. math.floor((prix % 10000) / 100) .. L["gold"])
				else
					print(L["CantRepair"])
				end
			elseif GearHelper.db.profile.autoRepair == 2 then
				if droitGuilde ~= nil and (droitGuilde == -1 or (droitGuilde > argentGuilde and argentGuilde > prix)) then
					RepairAllItems(true)
					print(GearHelper:ColorizeString(L["guildRepairCost"], "Pink") .. math.floor(prix / 10000) .. L["dot"] .. math.floor((prix % 10000) / 100) .. L["gold"])
				else
					if argentPossedee >= prix then
						RepairAllItems(false)
						print(GearHelper:ColorizeString(L["repairCost"], "Pink") .. math.floor(prix / 10000) .. L["dot"] .. math.floor((prix % 10000) / 100) .. L["gold"])
					else
						print(L["CantRepair"])
					end
				end
			end
		end
	end

	for bag = 0, 4 do
		for slot = 1, GetContainerNumSlots(bag) do
			if GetContainerItemID(bag, slot) ~= nil then
				local _, itemCount = GetContainerItemInfo(bag, slot)
				local id = GetContainerItemID(bag, slot)
				if id then
					local _, _, _, _, _, _, _, _, _, _, vendorPrice = GetItemInfo(id)
					if vendorPrice ~= nil then
						gagne = gagne + (vendorPrice * itemCount)
					end
				end
			end
		end
	end
end

-- local function Dot()
-- 	GearHelper:poseDot()
-- 	print("On a appelé posedot")
-- end
local function PlayerEnteringWorld()
	GearHelper:BenchmarkCountFuncCall("PlayerEnteringWorld")
	local used = false
	for i = 1, NUM_CHAT_WINDOWS do
		local _, _, _, _, _, _, _, _, _, uninteractable = GetChatWindowInfo(i)
		if (uninteractable) then
			SetChatWindowUninteractable(i, false)
			used = true
		end
	end
	if used then
		ReloadUI()
	end

	GearHelper:BuildCWTable()
	if GearHelper.db.profile.addonEnabled == true then
		GearHelper:sendAskVersion()
		GearHelper:ScanCharacter()
		GearHelper:poseDot()
		-- ContainerFrame1:HookScript("OnShow", Dot)
		if (not string.match(GearHelper.db.global.myNames, GetUnitName("player") .. ",")) then
			GearHelper.db.global.myNames = GearHelper.db.global.myNames .. GetUnitName("player") .. ","
		end
	end

	local _, instanceType, _, _, _, _, _, _, _, lfgDungeonsID = GetInstanceInfo()
	if "raid" == instanceType then
		if (nil == lfgDungeonsID) then
			lfrCheckIsChecked = false
			if (lfrCheckButton_GlobalName) then
				lfrCheckButton_GlobalName:Hide()
			end
			do
				return
			end
		end
		-- si le joueur est dans un groupe LFR afficher sous la minimap une case cochable permettant d'accepter automatiquement les appels

		lfrCheckButton = lfrCheckButton_GlobalName or CreateFrame("CheckButton", "lfrCheckButton_GlobalName", UIParent, "ChatConfigCheckButtonTemplate")
		lfrCheckButton:SetPoint("TOPRIGHT", -325, -50)
		lfrCheckButton_GlobalNameText:SetText(L["lfrCheckButtonText"])
		lfrCheckButton.tooltip = L["lfrCheckButtonTooltip"]
		lfrCheckButton:SetScript(
			"OnClick",
			function()
				lfrCheckIsChecked = lfrCheckButton:GetChecked()
			end
		)
		-- else
		lfrCheckButton_GlobalName:Show()
		lfrCheckButton_GlobalName:SetChecked(lfrCheckIsChecked)
	else
		lfrCheckIsChecked = false
		if (lfrCheckButton_GlobalName) then
			lfrCheckButton_GlobalName:Hide()
		end
	end
end

local function ChatMsgAddon(_, _, prefixMessage, message, _, sender)
	GearHelper:BenchmarkCountFuncCall("ChatMsgAddon")
	if prefixMessage ~= GearHelperVars.prefixAddon then
		do
			return
		end
	end
	if not GearHelper.db.profile.addonEnabled then
		do
			return
		end
	end

	local emetteur = ""
	if sender:find("-") then
		emetteur = sender:sub(0, (sender:find("-") - 1))
	else
		emetteur = sender
	end

	if emetteur == GetUnitName("player") then
		do
			return
		end
	end
	local prefVersion = message:sub(0, (message:find(";") - 1))
	if prefVersion == "answerVersion" then
		local vVersion = message:sub(message:find(";") + 1, #message)
		versionCible = vVersion
		GearHelper:receiveAnswer(vVersion, sender)
	end
	if prefVersion == "askVersion" then
		GearHelper:sendAnswerVersion()
	end
end

local function ItemPush(_, _, bag)
	GearHelper:BenchmarkCountFuncCall("ItemPush")
	if not GearHelper.db.profile.autoEquipLooted.actual then
		do
			return
		end
	end

	local theBag = bag
	if bag == 23 then
		theBag = 4
	elseif bag == 22 then
		theBag = 3
	elseif bag == 21 then
		theBag = 2
	elseif bag == 20 then
		theBag = 1
	end
	GearHelper:equipItem(theBag)
end

local function QuestComplete()
	GearHelper:BenchmarkCountFuncCall("QuestComplete")
	GearHelper.GetQuestRewardCoroutine =
		coroutine.create(
		function()
			GearHelper:GetQuestReward()
		end
	)
	coroutine.resume(GearHelper.GetQuestRewardCoroutine)
end

local function MerchantClosed()
	GearHelper:BenchmarkCountFuncCall("MerchantClosed")
	if not GearHelper.db.profile.sellGreyItems then
		do
			return
		end
	end

	local argentFin = 0
	for bag = 0, 4 do
		for slot = 1, GetContainerNumSlots(bag) do
			if GetContainerItemID(bag, slot) ~= nil then
				local _, itemCount = GetContainerItemInfo(bag, slot)
				local id = GetContainerItemID(bag, slot)
				if id then
					local _, _, _, _, _, _, _, _, _, _, vendorPrice = GetItemInfo(id)
					argentFin = argentFin + (vendorPrice * itemCount)
				end
			end
		end
	end
	if (gagne - argentFin > 0) then
		print(GearHelper:ColorizeString(L["moneyEarned"], "LightGreen") .. math.floor((gagne - argentFin) / 10000) .. L["dot"] .. math.floor(((gagne - argentFin) % 10000) / 100) .. L["gold"])
		gagne = 0
	end
end

local function BagUpdate()
	GearHelper:BenchmarkCountFuncCall("BagUpdate")
	if time() - lastBagUpdateEvent < 2 then
		return
	end
	lastBagUpdateEvent = time()
	if not GearHelperVars.charInventory["MainHand"] then
		do
			return
		end
	end
	if GearHelperVars.charInventory["MainHand"] == "" then
		do
			return
		end
	end
	-- Random check to verify that charInventory is initialized because BagUpdate is fired before PlayerEnteringWorld
	GearHelper:ScanCharacter()
	GearHelper:poseDot()
end

local function ActiveTalentGroupChanged()
	GearHelper:BenchmarkCountFuncCall("ActiveTalentGroupChanged")
	if not GearHelper.db.profile.autoEquipWhenSwitchSpe then
		GearHelper.cwTable.args["NoxGroup"].name = "Noxxic " .. (GetSpecialization() and select(2, GetSpecializationInfo(GetSpecialization())) or "None")
		do
			return
		end
	end

	GearHelperVars.waitSpeTimer = time()
	GearHelperVars.waitSpeFrame:Show()
	GearHelper:equipItem(0)
	GearHelper:equipItem(1)
	GearHelper:equipItem(2)
	GearHelper:equipItem(3)
	GearHelper:equipItem(4)
end

local function ChatMsgChannel(_, _, msg, sender, lang, channel)
	GearHelper:BenchmarkCountFuncCall("ChatMsgChannel")
	if not GearHelper.db.profile.autoInvite or not msg then
		GearHelper:showMessageSMN(channel, sender, msg)
		do
			return
		end
	end

	local playerIsNotMe = not string.find(sender, GetUnitName("player"))
	if msg:lower() == GearHelper.db.profile.inviteMessage:lower() and playerIsNotMe and GetNumGroupMembers() == 5 then
		ConvertToRaid()
		InviteUnit(sender)
	elseif msg:lower() == GearHelper.db.profile.inviteMessage:lower() and playerIsNotMe then
		InviteUnit(sender)
	end

	GearHelper:showMessageSMN(channel, sender, msg)
end

local function ChatMsgWhisper(_, _, msg, sender)
	GearHelper:BenchmarkCountFuncCall("ChatMsgWhisper")
	if GearHelper.db.profile.autoInvite and msg ~= nil then
		local playerIsNotMe = not string.find(sender, GetUnitName("player"))
		if msg:lower() == GearHelper.db.profile.inviteMessage:lower() and playerIsNotMe and GetNumGroupMembers() == 5 then
			ConvertToRaid()
			InviteUnit(sender)
		elseif msg:lower() == GearHelper.db.profile.inviteMessage:lower() and playerIsNotMe then
			InviteUnit(sender)
		end
	end
	if (GearHelper.db.profile.whisperAlert) then
		PlaySoundFile("Interface\\AddOns\\GearHelper\\Textures\\whisper.ogg", "Master")
	end
end

local function ChatMsgLoot(_, _, message, language, sender, channelString, target, flags, unknown1, channelNumber, channelName, unknown2, counter)
	GearHelper:BenchmarkCountFuncCall("ChatMsgLoot")
	GearHelper:CreateLinkAskIfHeNeeds(0, message, sender, language, channelString, target, flags, unknown1, channelNumber, channelName, unknown2, counter)
end

local function ChatMsgBattleground(_, _, msg, sender, lang, channel)
	GearHelper:BenchmarkCountFuncCall("ChatMsgBattleground")
	GearHelper:showMessageSMN("BG", sender, msg)
end

local function ChatMsgBattlegroundLeader(_, _, msg, sender, lang, channel)
	GearHelper:BenchmarkCountFuncCall("ChatMsgBattlegroundLeader")
	GearHelper:showMessageSMN("BG", sender, msg)
end

local function ChatMsgEmote(_, _, msg, sender, lang, channel)
	GearHelper:BenchmarkCountFuncCall("ChatMsgEmote")
	GearHelper:showMessageSMN("Emote", sender, msg)
end

local function ChatMsgGuild(_, _, msg, sender, lang, channel)
	GearHelper:BenchmarkCountFuncCall("ChatMsgGuild")
	GearHelper:showMessageSMN("Guild", sender, msg)
end

local function ChatMsgOfficer(_, _, msg, sender, lang, channel)
	GearHelper:BenchmarkCountFuncCall("ChatMsgOfficer")
	GearHelper:showMessageSMN("Officer", sender, msg)
end

local function ChatMsgParty(_, _, msg, sender, lang, channel)
	GearHelper:BenchmarkCountFuncCall("ChatMsgParty")
	GearHelper:showMessageSMN("Party", sender, msg)
end

local function ChatMsgPartyLeader(_, _, msg, sender, lang, channel)
	GearHelper:BenchmarkCountFuncCall("ChatMsgPartyLeader")
	GearHelper:showMessageSMN("Party", sender, msg)
end

local function ChatMsgRaid(_, _, msg, sender, lang, channel)
	GearHelper:BenchmarkCountFuncCall("ChatMsgRaid")
	GearHelper:showMessageSMN("Raid", sender, msg)
end

local function ChatMsgRaidLeader(_, _, msg, sender, lang, channel)
	GearHelper:BenchmarkCountFuncCall("ChatMsgRaidLeader")
	GearHelper:showMessageSMN("Raid", sender, msg)
end

local function ChatMsgRaidWarning(_, _, msg, sender, lang, channel)
	GearHelper:BenchmarkCountFuncCall("ChatMsgRaidWarning")
	GearHelper:showMessageSMN("Raid_warning", sender, msg)
end

local function ChatMsgSay(_, _, msg, sender, lang, channel)
	GearHelper:BenchmarkCountFuncCall("ChatMsgSay")
	GearHelper:showMessageSMN("Say", sender, msg)
end

local function ChatMsgYell(_, _, msg, sender, lang, channel)
	GearHelper:BenchmarkCountFuncCall("ChatMsgYell")
	GearHelper:showMessageSMN("Yell", sender, msg)
end

local function UnitInventoryChanged(_, _, joueur)
	GearHelper:BenchmarkCountFuncCall("UnitInventoryChanged")
	if not GearHelper.db.profile.addonEnabled then
		do
			return
		end
	end
	if joueur ~= "player" then
		do
			return
		end
	end

	GearHelper:ScanCharacter()

	if not GetInventoryItemLink("player", GetInventorySlotInfo("MainHandSlot")) then
		do
			return
		end
	end

	local _, _, _, _, _, _, subclass = GetItemInfo(GetInventoryItemLink("player", GetInventorySlotInfo("MainHandSlot")))
	if subclass == L["cannapeche"] then
		GearHelper.db.profile.autoEquipLooted.previous = GearHelper.db.profile.autoEquipLooted.actual
		GearHelper.db.profile.autoEquipLooted.actual = false
	else
		GearHelper.db.profile.autoEquipLooted.actual = GearHelper.db.profile.autoEquipLooted.previous
	end
end

local function QuestTurnedIn()
	GearHelper:BenchmarkCountFuncCall("QuestTurnedIn")
	if not GearHelper.db.profile.autoEquipLooted.actual then
		do
			return
		end
	end

	GearHelperVars.waitSpeTimer = time()
	GearHelperVars.waitSpeFrame:Show()
end

local function GetItemInfoReceived(_, _, item)
	GearHelper:BenchmarkCountFuncCall("GetItemInfoReceived")
	if GearHelper.db.global.itemWaitList[item] then
		local slotName = GearHelper.db.global.itemWaitList[item]
		GearHelper.db.global.itemWaitList[item] = nil
		GearHelperVars.charInventory[string.sub(slotName, 1, -5)] = GearHelper:GetEquippedItemLink(GetInventorySlotInfo(slotName), slotName)
	end
	if item ~= nil then
		if GearHelper.idNilGetQuestReward ~= nil then
			if item == GearHelper.idNilGetQuestReward then
				GearHelper:Print(tostring(item) .. " était nil")
				coroutine.resume(GearHelper.GetQuestRewardCoroutine)
			end
		end
	end

	if (InspectPaperDollItemsFrame) then
		NotifyInspect("target")
	end
end

local function ReadyCheck(_, _)
	GearHelper:BenchmarkCountFuncCall("ReadyCheck")
	local players = GetHomePartyInfo()
end

local function LfgUpdate(_, _)
	GearHelper:BenchmarkCountFuncCall("LfgUpdate")
	GearHelper:UpdateGHLfrButton()
end

local function PlayerLogin(_, _)
	GearHelper:BenchmarkCountFuncCall("PlayerLogin")
	-- Si la frame recherche donjon est ouverte et que la fonction de selection de donjon est dispo (sur la page lfr en gros)
	if RaidFinderQueueFrame and RaidFinderQueueFrame_SetRaid then
		local function LfrFrameShow(frame)
			GearHelper:BenchmarkCountFuncCall("LfrFrameShow")
			if not GearHelper.db.profile.bossesKilled then
				do
					return
				end
			end
			GearHelper:CreateLfrButtons(frame)
			GearHelper:UpdateButtonsAndTooltips(frame)
			GearHelper:UpdateGHLfrButton()
			GearHelper:UpdateSelecCursor()
			GearHelper:RegisterEvent("LFG_UPDATE")
			GearHelper.LFG_UPDATE = GearHelper.UpdateGHLfrButton
		end
		local function LfrFrameHide()
			GearHelper:BenchmarkCountFuncCall("LfrFrameHide")
			GearHelper:HideLfrButtons()
			GearHelper:UnregisterEvent("LFG_UPDATE")
		end

		RaidFinderQueueFrame:HookScript("OnShow", LfrFrameShow)
		RaidFinderQueueFrame:HookScript("OnHide", LfrFrameHide)
		hooksecurefunc("RaidFinderQueueFrame_SetRaid", GearHelper.UpdateSelecCursor)
	end

	-- Si la page du personnage s'affiche, on affiche l'ilvl
	if (PaperDollItemsFrame) then
		GearHelper:AddIlvlOnCharFrame()
	end
end

local function InspectReady(_, _, target)
	GearHelper:BenchmarkCountFuncCall("InspectReady")
	if GearHelper.db.profile.inspectAin.waitingIlvl then ---------------- /GH AIN AVEC UN MESSAGE SPÉCIALE SI L'ILVL DE L'OBJET LOOT EST MOINS BON QUE CELUI ÉQUIPPÉ PAR CELUI QUI L'A LOOT
		local itemLoot = GearHelper.db.profile.inspectAin.linkItemReceived
		local itemLootTable = GearHelper:GetItemByLink(itemLoot)
		local itemLootEquipLoc = GearHelper.db.global.equipLocInspect[itemLootTable.equipLoc]

		if (itemLootEquipLoc ~= 11 and itemLootEquipLoc ~= 12 and itemLootEquipLoc ~= 13 and itemLootEquipLoc ~= 14) then
			local itemEquipped = GetInventoryItemLink(target, itemLootEquipLoc)
			if (not itemEquipped) then
				do
					return
				end
			end
			local itemEquippedTable = GearHelper:GetItemByLink(itemEquipped)
			local itemEquippedIlvl = itemEquippedTable.iLvl
			local itemLootIlvl = itemLootTable.iLvl

			print("-------------")
			print("item équippé par " .. GearHelper.db.profile.inspectAin.target .. " : " .. itemEquipped)
			print("item reçu : " .. itemLoot)
			if (itemEquippedIlvl >= itemLootIlvl) then
				print("Vous pouvez essayer de demander cet objet")
			else
				print("L'item loot à un meilleur ilvl, + de chances de refus")
			end
		end

		GearHelper.db.profile.inspectAin.waitingIlvl = false
		GearHelper.db.profile.inspectAin.linkItemReceived = nil
		GearHelper.db.profile.inspectAin.message = nil
		GearHelper.db.profile.inspectAin.target = nil

		ClearInspectPlayer()
	elseif (InspectPaperDollItemsFrame) then -------------------- AFFICHE L'ILVL DES ITEMS D'UN JOUEUR SUR LA FICHE D'INSPECTION
		GearHelper:AddIlvlOnInspectFrame(target)
	else ------------------- AFFICHE L'ILVL MOYEN D'UN MEC SUR SON TOOLTIP
		if not GameTooltip:IsVisible() then
			do
				return
			end
		end

		local arrayIlvl = {}
		for i = 1, 19 do
			local itemLink = GetInventoryItemLink("mouseover", i)
			if (itemLink) then
				local itemScan = GearHelper:GetItemByLink(itemLink)
				local itemLvl, equipLoc = itemScan.iLvl, itemScan.equipLoc
				if equipLoc ~= nil then
					arrayIlvl[equipLoc] = itemLvl
					table.insert(arrayIlvl, itemLvl)
				end
			end
		end
		local ilvlAverage = 0
		local itemCount = 0
		table.foreach(
			arrayIlvl,
			function(equipLoc, ilvl)
				if (equipLoc ~= "INVTYPE_TABARD" and equipLoc ~= "INVTYPE_BODY") then
					ilvlAverage = ilvlAverage + ilvl
					itemCount = itemCount + 1
				end
			end
		)
		if (itemCount ~= 0) then
			GameTooltip:AddLine(L["ilvlInspect"] .. tostring(math.floor((ilvlAverage / itemCount) + .5)))
		end

		ClearInspectPlayer()
		GameTooltip:Show()
	end
end

local function UpdateMouseOverUnit()
	GearHelper:BenchmarkCountFuncCall("UpdateMouseOverUnit")
	if not CanInspect("mouseover") then
		do
			return
		end
	end
	if not CheckInteractDistance("mouseover", 1) then
		do
			return
		end
	end

	NotifyInspect("mouseover")
end

local function ReadyCheck()
	GearHelper:BenchmarkCountFuncCall("ReadyCheck")
	if lfrCheckIsChecked then
		ConfirmReadyCheck(1)
		ReadyCheckFrame:Hide()
		print("Ready check accepted")
		UIErrorsFrame:AddMessage("Ready check accepted", 0.0, 1.0, 0.0, 80)
	end
end

GearHelper:RegisterEvent("ADDON_LOADED", AddonLoaded, ...)
GearHelper:RegisterEvent("MERCHANT_SHOW", OnMerchantShow)
GearHelper:RegisterEvent("PLAYER_ENTERING_WORLD", PlayerEnteringWorld)
GearHelper:RegisterEvent("CHAT_MSG_ADDON", ChatMsgAddon, ...)
GearHelper:RegisterEvent("ITEM_PUSH", ItemPush, ...)
GearHelper:RegisterEvent("QUEST_COMPLETE", QuestComplete)
GearHelper:RegisterEvent("MERCHANT_CLOSED", MerchantClosed)
GearHelper:RegisterEvent("BAG_UPDATE", BagUpdate)
GearHelper:RegisterEvent("ACTIVE_TALENT_GROUP_CHANGED", ActiveTalentGroupChanged)
GearHelper:RegisterEvent("CHAT_MSG_CHANNEL", ChatMsgChannel, ...)
GearHelper:RegisterEvent("CHAT_MSG_WHISPER", ChatMsgWhisper, ...)
GearHelper:RegisterEvent("CHAT_MSG_LOOT", ChatMsgLoot, ...)
GearHelper:RegisterEvent("CHAT_MSG_EMOTE", ChatMsgEmote, ...)
GearHelper:RegisterEvent("CHAT_MSG_GUILD", ChatMsgGuild, ...)
GearHelper:RegisterEvent("CHAT_MSG_OFFICER", ChatMsgOfficer, ...)
GearHelper:RegisterEvent("CHAT_MSG_PARTY", ChatMsgParty, ...)
GearHelper:RegisterEvent("CHAT_MSG_PARTY_LEADER", ChatMsgPartyLeader, ...)
GearHelper:RegisterEvent("CHAT_MSG_RAID", ChatMsgRaid, ...)
GearHelper:RegisterEvent("CHAT_MSG_RAID_LEADER", ChatMsgRaidLeader, ...)
GearHelper:RegisterEvent("CHAT_MSG_RAID_WARNING", ChatMsgRaidWarning, ...)
GearHelper:RegisterEvent("CHAT_MSG_SAY", ChatMsgSay, ...)
GearHelper:RegisterEvent("CHAT_MSG_YELL", ChatMsgYell, ...)
GearHelper:RegisterEvent("UNIT_INVENTORY_CHANGED", UnitInventoryChanged, ...)
GearHelper:RegisterEvent("QUEST_TURNED_IN", QuestTurnedIn)
GearHelper:RegisterEvent("GET_ITEM_INFO_RECEIVED", GetItemInfoReceived, ...)
GearHelper:RegisterEvent("PLAYER_LOGIN", PlayerLogin, ...)
GearHelper:RegisterEvent("LFG_UPDATE", LfgUpdate, ...)
GearHelper:RegisterEvent("INSPECT_READY", InspectReady, ...)
GearHelper:RegisterEvent("UPDATE_MOUSEOVER_UNIT", UpdateMouseOverUnit, ...)
GearHelper:RegisterEvent("READY_CHECK", ReadyCheck, ...)
